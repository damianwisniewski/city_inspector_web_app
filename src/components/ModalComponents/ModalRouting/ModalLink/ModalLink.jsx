import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ModalContext } from '../ModalRouter/ModalRouter'

class ModalLink extends Component {
	static contextType = ModalContext

	render() {
		const { children, to, activeClassName, className, ...props } = this.props
		const { handleOpenModal, componentName } = this.context

		return (
			<button
				{...props}
				data-to={to}
				onClick={handleOpenModal}
				className={`${className} ${activeClassName && to === componentName ? activeClassName : ''}`}
			>
				{children}
			</button>
		)
	}
}

ModalLink.propTypes = {
	children: PropTypes.node,
	to: PropTypes.string.isRequired,
}

export default ModalLink
