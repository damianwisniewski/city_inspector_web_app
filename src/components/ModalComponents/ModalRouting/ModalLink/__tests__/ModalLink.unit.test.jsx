import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import ModalLink from '../ModalLink'

configure({ adapter: new Adapter() })

describe('<ModalLink />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let ModalLinkComponent
	const textContent = 'Example text'
	const to = 'FakeModal'
	const activeClassName = 'FakeCssClass'
	const className = 'FakeCssClass_2'

	test('should render correctly', () => {
		ModalLinkComponent = shallow(
			<ModalLink to={to} activeClassName={activeClassName} className={className}>
				{textContent}
			</ModalLink>,
		)

		expect(toJson(ModalLinkComponent)).toMatchSnapshot()
	})

	//TODO: tests related to context
})
