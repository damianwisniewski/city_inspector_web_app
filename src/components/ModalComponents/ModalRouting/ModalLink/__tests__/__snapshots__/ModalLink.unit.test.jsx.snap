// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalLink /> should render correctly 1`] = `
<button
  className="FakeCssClass_2 "
  data-to="FakeModal"
>
  Example text
</button>
`;
